src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/SchedulerTimestampRepositoryExtensionsImpl.java

```diff
@@ -0,0 +1,60 @@
+package uk.nhs.digital.nhsconnect.nhais.repository;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+import java.time.LocalDateTime;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.stereotype.Repository;
+
+import org.bson.Document;
+import com.mongodb.client.result.UpdateResult;
+
+@Repository
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
+public class SchedulerTimestampRepositoryExtensionsImpl implements SchedulerTimestampRepositoryExtensions {
+
+    private static final String MESH_TIMESTAMP_COLLECTION_NAME = "schedulerTimestamp";
+    private static final String SCHEDULER_TYPE = "schedulerType";
+    private static final String TIMESTAMP_FIELD_NAME = "timestamp";
+
+    private final MongoOperations mongoOperations;
+
+    @Override
+    public boolean updateTimestamp(String schedulerType, LocalDateTime timestamp, long seconds) {
+        var query = query(where(TIMESTAMP_FIELD_NAME).lt(LocalDateTime.now().minusSeconds(seconds))
+            .and(SCHEDULER_TYPE).is(schedulerType));
+
+        var update = Update.update(TIMESTAMP_FIELD_NAME, timestamp)
+            .set(SCHEDULER_TYPE, schedulerType);
+
+      if (collectionIsNotEmpty()) {
+          UpdateResult result = mongoOperations.updateFirst(query, update, MESH_TIMESTAMP_COLLECTION_NAME);
+
+          return updateSuccessful(result);
+        } else {
+            LOGGER.info("{} collection does not exist or it is empty. Document with timestamp will be created", MESH_TIMESTAMP_COLLECTION_NAME);
+            Document document = new Document();
```

Comment:
Can't you use `SchedulerTimestamp` instead of raw `Document` ?

Reply:
ok